addons/account_edi_ubl/models/account_edi_format.py

```diff
@@ -37,19 +37,19 @@ def _get_ubl_namespaces():
             return namespaces
 
         namespaces = _get_ubl_namespaces()
-        if not invoice:
-            invoice = self.env['account.move'].create({})
 
-        elements = tree.xpath('//cbc:InvoiceTypeCode', namespaces=namespaces)
-        if elements:
-            type_code = elements[0].text
-            move_type = 'in_refund' if type_code == '381' else 'in_invoice'
+        journal = invoice._context.get('default_journal_id')
+        if journal:
+            move_type = 'out_invoice' if journal.type == 'sale' else 'in_invoice'
         else:
-            move_type = 'in_invoice'
+            move_type = invoice._context.get('default_move_type', 'in_invoice')
+            journal = invoice.with_context(default_move_type=move_type)._get_default_journal()
```

Comment:
This is the reason why we splitted the creation/update at the first time. Why not managing that in `_create_invoice_from_xml_tree` and `_update_invoice_from_xml_tree` instead ?

Reply:
`_create_invoice_from_xml_tree` should check edi_format.code before processing, since we are in a generic method, this could be called by multiple UBL implementations. I created a helper `_create_invoice_from_ubl` that does that.